feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/newApp/NewSubstrateLedgerApplication.kt

```diff
@@ -0,0 +1,113 @@
+package io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.newApp
+
+import android.util.Log
+import io.novafoundation.nova.common.utils.chunked
+import io.novafoundation.nova.common.utils.littleEndianBytes
+import io.novafoundation.nova.feature_ledger_api.sdk.application.substrate.LedgerSubstrateAccount
+import io.novafoundation.nova.feature_ledger_api.sdk.application.substrate.SubstrateLedgerApplication
+import io.novafoundation.nova.feature_ledger_api.sdk.device.LedgerDevice
+import io.novafoundation.nova.feature_ledger_api.sdk.transport.LedgerTransport
+import io.novafoundation.nova.feature_ledger_api.sdk.transport.send
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.SubstrateLedgerAppCommon
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.SubstrateLedgerAppCommon.CHUNK_SIZE
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.SubstrateLedgerAppCommon.defaultCryptoScheme
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.SubstrateLedgerAppCommon.encodeDerivationPath
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.SubstrateLedgerAppCommon.parseAccountResponse
+import io.novafoundation.nova.feature_ledger_impl.sdk.application.substrate.newApp.metadata.MetadataShortenerService
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import io.novasama.substrate_sdk_android.encrypt.SignatureWrapper
+import io.novasama.substrate_sdk_android.extensions.toHexString
+import io.novasama.substrate_sdk_android.runtime.extrinsic.signer.SignerPayloadExtrinsic
+import io.novasama.substrate_sdk_android.runtime.extrinsic.signer.encodedSignaturePayload
+
+abstract class NewSubstrateLedgerApplication(
+    private val transport: LedgerTransport,
+    private val chainRegistry: ChainRegistry,
+    private val metadataShortenerService: MetadataShortenerService
+) : SubstrateLedgerApplication {
+
+    abstract val cla: UByte
+
+    abstract suspend fun getDerivationPath(chainId: ChainId, accountIndex: Int): String
+
+    abstract suspend fun getDerivationPath(metaId: Long, chainId: ChainId): String
+
+    override suspend fun getAccount(device: LedgerDevice, chainId: ChainId, accountIndex: Int, confirmAddress: Boolean): LedgerSubstrateAccount {
+        val displayVerificationDialog = SubstrateLedgerAppCommon.DisplayVerificationDialog.fromBoolean(confirmAddress)
+
+        val derivationPath = getDerivationPath(chainId, accountIndex)
+        val encodedDerivationPath = encodeDerivationPath(derivationPath)
+
+        val chain = chainRegistry.getChain(chainId)
+        val ss58Prefix = chain.addressPrefix.toShort()
+
+        val payload = encodedDerivationPath + ss58Prefix.littleEndianBytes
+
```

Comment:
Note application level uses little endian encoding compared to big endian on transport level